src/prefect/blocks/core.py

```diff
@@ -220,6 +227,24 @@ def get_block_capabilities(cls) -> FrozenSet[str]:
             }
         )
 
+    @classmethod
+    def _get_current_package_version(cls):
+        current_module = inspect.getmodule(cls)
+        if current_module:
+            top_level_module = sys.modules[
+                current_module.__name__.split(".")[0] or "__main__"
+            ]
+            try:
+                return str(top_level_module.__version__)
```

Comment:
we might want to use `packaging.Version` here to parse version numbers just so there's a bit more uniformity (let's us also squash dirty version numbers that versioneer can generate etc if we want)